phoenix-core/src/main/java/org/apache/phoenix/execute/MutationState.java

```diff
@@ -1199,7 +1201,20 @@ private void filterIndexCheckerMutations(Map<TableInfo, List<Mutation>> mutation
             Entry<TableInfo, List<Mutation>> pair = mapIter.next();
             TableInfo tableInfo = pair.getKey();
             if (tableInfo.getOrigTableRef().getTable().isImmutableRows() && IndexUtil.isGlobalIndexCheckerEnabled(connection, tableInfo.getHTableName())) {
-                PTable table = PhoenixRuntime.getTable(connection, tableInfo.getHTableName().getString());
+                PTable table;
+                if (MetaDataUtil.isViewIndex(tableInfo.getHTableName().getString())) {
+                    PTable origLogicalTable = tableInfo.getOrigTableRef().getTable();
+                    if (origLogicalTable.getTenantId() == null) {
+                        //Global view index
+                        table = origLogicalTable.getIndexes().get(0);
```

Comment:
Does it matter if there is more than one index on the view?

Reply:
I modified to use MetaDataUtil method to get the logical table.